plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/connection/ConnectionsRegistry.java

```diff
@@ -150,54 +142,37 @@ public boolean remove(IConnection connection) {
 		}
 	}
 
-	private void fireChange(IConnection  connection, int event) {
+	private void fireChange(IConnection connection, int event) {
 		fireChange(connection, event, null, null, null);
 	}
-	private void fireChange(IConnection  connection, int event, String property, Object oldValue, Object newValue) {
+
+	private void fireChange(IConnection connection, int event, String property, Object oldValue, Object newValue) {
 		if (connection == null) {
 			return;
 		}
-		Job job = new AbstractDelegatingMonitorJob("Connection Changed") {
-		
-			@Override
-			protected IStatus doRun(IProgressMonitor monitor) {
-				try {
-					Iterator<IConnectionsRegistryListener> i = getListeners().iterator();
-					while (i.hasNext()) {
-						IConnectionsRegistryListener l = i.next();
-						switch (event) {
-						case ADDED:
-							l.connectionAdded(connection);
-							break;
-						case REMOVED:
-							l.connectionRemoved(connection);
-							break;
-						case CHANGED:
-							l.connectionChanged(connection, property, oldValue, newValue);
-							break;
-							
-						default:
-							break;
-						}
-					}
-				}catch(Exception e) {
-					return new Status(IStatus.ERROR,OpenShiftCommonCoreActivator.PLUGIN_ID,"Exception while notifying of a connection change",e);
-				}	
-				return Status.OK_STATUS;
+
+		for (IConnectionsRegistryListener l : getListeners()) {
+			switch (event) {
+			case ADDED:
+				l.connectionAdded(connection);
+				break;
+			case REMOVED:
+				l.connectionRemoved(connection);
+				break;
+			case CHANGED:
+				l.connectionChanged(connection, property, oldValue, newValue);
+				break;
 			}
-		};
-		job.schedule();
+		}
 	}
 
 	public IConnection getRecentConnection() {
 		return recentConnection;
 	}
-	
+
 	@SuppressWarnings("unchecked")
 	public <T extends IConnection> T getRecentConnection(Class<T> clazz) {
-		if (recentConnection == null
-				|| clazz == null
-				|| !clazz.isAssignableFrom(recentConnection.getClass())) {
+		if (recentConnection == null || clazz == null || !clazz.isAssignableFrom(recentConnection.getClass())) {
```

Comment:
I'd love if we could keep every condition in this "if" in its own line, it helps in reading the code and grasping every single condition.